components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt

```diff
@@ -151,7 +159,8 @@ open class PlacesHistoryStorage(
 
     override fun getAutocompleteSuggestion(query: String): HistoryAutocompleteResult? {
         val url = handlePlacesExceptions("getAutoCompleteSuggestions", default = null) {
-            places.reader().matchUrl(query)
+            autocompleteReader.interrupt()
```

Comment:
@Mugurell So before this fix here we didn't interrupt autocompletes. We only interrupted history suggestions. How did sharing the reader cause us to lose history suggestions?

Reply:
I assume parallel queries on the same reader - for reading suggestions and for reading autocomplete and a race between these requests which would explain the intermittent nature of the issue. 
Given that both queries happen in `PlacesHistoryStorage` they will share the same reader. 
If [`cancelReads`](https://github.com/mozilla-mobile/android-components/blob/52e2fc78d6b1e6ea1ad977e47fb3d8e09e3792f0/components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/HistoryStorageSuggestionProvider.kt#L57) from before new suggestions requests happen after the `getAutocompleteSuggestion` gets to make a request to `places.reader().matchUrl` but before this completes then `places.reader().matchUrl` would also be interrupted [and so it would return null](https://github.com/mozilla-mobile/android-components/blob/1b490c833c9935a250b7b4c1e29eb7f272d3e75f/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt#L152-L155).

Reply:
@Mugurell Yes, I'd understand that code path, but that would mean we'd lose autocomplete suggestions, not history search suggestions. The way I understand the ticket is that search suggestions are missing from the awesomebar, right?

Reply:
(If it's speculative and you can't repro anymore, but we don't know exactly why, I am also fine landing this and asking folks for feedback.)

Reply:
Sorry, didn't touch the "history suggestions" part.
Based on the user reports from https://github.com/mozilla-mobile/fenix/issues/26145 the issue is with autocomplete not showing result from history but from the list of shipped domains.
And indeed while debugging I saw that although there should be matching history entries in this call
https://github.com/mozilla-mobile/android-components/blob/19d85f964578a32f25f2c96226de48af52a56da0/components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarAutocompleteFeature.kt#L37-L42
`historyResults` would be null and so we would show `domainResults`.